### ISSUE TITLE
More warnings that can't be caught

### ISSUE THREAD

**Issue by pitakakariki** [lme4/lme4#331]
NB: I don't have enough time today to make this report reproducible, but I can come back and do that later this week if you think that would be helpful.

This is really a bug in R, rather than lme4. It's possible to put a workaround into lme4, but it might be more sensible to escalate this to R itself.

I only ran into this because I was using a "year" variable that wasn't centred as a fixed effect, so it's unlikely to cause anyone too many headaches.

---

This is another case of https://github.com/lme4/lme4/issues/304. See also [this Stack Overflow post](http://stackoverflow.com/questions/20268021/how-to-catch-warnings-sent-during-s4-method-selection).

In the definition for `summary.merMod`:

```
    coefs <- cbind("Estimate" = coefs,
               "Std. Error" = sqrt(diag(vcov(object,
               use.hessian = use.hessian))))
```

Because `diag` is an S4 method, any warnings from vcov won't be caught by calling handlers.

**Comment by pitakakariki** [comment]
I ran into this bug again today. The underlying bug in R is at least two years old! Anyone here have enough influence to get its priority increased?

**Comment by bbolker** [comment]
can you provide a link to a corresponding R bug report?

**Comment by pitakakariki** [comment]
I think it's this one: https://bugs.r-project.org/bugzilla3/show_bug.cgi?id=16111

**Comment by pitakakariki** [comment]
Thanks @mmaechler for chasing this up. Do you want to close this as an issue here, or leave it until the underlying R bug is fixed?

**Comment by mmaechler** [comment]
> Do you want to close this as an issue here, or leave it until the
    > underlying R bug is fixed?

I'd keep the issue open.  After having browsed and partly read
the C code in R,  I'm not 100% sure that our issue is exactly
corresponding to the R bug mentioned, or if it is just related.

Also it is not quite clear which future versions of R will have
this fixed  (R >= 3.3.0  for sure).

**Comment by pitakakariki** [comment]
The underlying R bug has been fixed.
